app/controllers/reports_controller.rb

```diff
@@ -118,20 +118,21 @@ def inspection_rate_report
     authorize :report, :inspection_rate_report?
     
     if params[:commit]
-      zone_id, building_id, start_time, end_time = collect_form_params
+      zone_id, building_id, start_time, end_time, archived = collect_form_params
 
-      rooms = Room.active
-                  .joins(floor: { building: :zone }).joins(:room_states)
+      rooms = Room.left_outer_joins(floor: { building: :zone })
+                  .joins(:room_states)
+                  .where(archived: archived)
                   .where(buildings: { id: building_id, zone_id: zone_id })
                   .where(room_states: { updated_at: start_time..end_time })
                   .group('rooms.id')
                   .select('rooms.*')
                   .select('COUNT(room_states.id) AS room_check_count')
                   .order('room_check_count DESC')
 
-      rooms_no_room_state = Room.active
-                                .left_outer_joins(:room_states)
-                                .joins(floor: { building: :zone })
+      rooms_no_room_state = Room.left_outer_joins(floor: { building: :zone })
+                                .joins(:room_states)
```

Comment:
For the inspection_rate_report rooms_no_room_state query has rooms that were never checked (have no room_state; this is why there is a separate query for these rooms). In the report's result, these rooms have 0.0% Inspection Rate. These rooms would never be archived so we don't need to run this query for archived data. And with `.where(archived: archived)` the query returns zero records even if there are rooms without room_states.

I suggest to do the following for this report

```
if archived
     rooms_no_room_state = []
else
     rooms_no_room_state = Room.active
                                .left_outer_joins(:room_states)
                                .joins(floor: { building: :zone })
                                .where(buildings: { id: building_id, zone_id: zone_id })
                                .where(room_states: { id: nil })
                                .where.not(id: rooms.map(&:id))
                                .select('rooms.*')
                                .select('0 AS room_check_count')
                                .order('rooms.room_number')
 end
```